draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -379,23 +380,41 @@ Algorithms in this document make use of utility functions described below.
 - round(x): floor(ceil(2\*x)/2), i.e. rounding to the nearest
   integer, rounding down when the remainder is precisely 0.5.
 
-- HashToBase(H,x): H(x)[0:round(log2(p))] mod p.
-  This method is parametrized by the cryptographic hash function H, which
-  outputs at least round(log2(p)) bits, where p is the prime order of base
-  field Fp. The function first computes the bitstring H(x), and truncates this
-  down to round(log2(p)) bits. The result of this is converted to an integer,
-  and reduced modulo p to give an element of Fp. We suggest a more detailed
-  algorithm in {{hashtobase}}.
+- HashToBase(x, i).
+  This method is parametrized by p and H, where p is the prime order of
+  the base field Fp, and H is a cryptographic hash function which
+  outputs at least round(log2(p)) bits.
+  The function first hashes x, truncates the result, and converts to an integer,
+  and reduced modulo p to give an element of Fp. The function returns
+  this element of Fp, along with an additional bit, which can be used for
+  choosing square roots.
```

Comment:
I am not completely sure whether the hash output must be truncated. Since all the bits of the hash must contribute _in some manner_ to the value X in Fp.